### PR TITLE
Extract optimizer configs into OptimizerConfig

### PR DESCRIPTION
**Pull request by raunaqmorarka**
## Description

There are a significant number of properties
related to optimizer rules in FeaturesConfig.
Extracting these to OptimizerConfig makes FeaturesConfig
smaller and provides a convenient grouping of optimizer
related flags in one place.

> Is this change a fix, improvement, new feature, refactoring, or other?

refactoring

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

engine configs refactoring
> How would you describe this change to a non-technical end user or system administrator?

no user facing change
## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/SystemSessionProperties.java [hunk 1]
```diff
@@ -158,7 +159,15 @@
...
-        this(new QueryManagerConfig(), new TaskManagerConfig(), new MemoryManagerConfig(), new FeaturesConfig(), new NodeMemoryConfig(), new DynamicFilterConfig(), new NodeSchedulerConfig());
+        this(
+                new QueryManagerConfig(),
+                new TaskManagerConfig(),
+                new MemoryManagerConfig(),
+                new FeaturesConfig(),
+                new NodeMemoryConfig(),
+                new DynamicFilterConfig(),
+                new NodeSchedulerConfig(),
+                new OptimizerConfig());
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/SystemSessionProperties.java:170]: nit: i'd add this after FeaturesConfig, since they remain related

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
`ci / test (plugin/trino-kudu) (pull_request) Failing after 7m — test (plugin/trino-kudu)`

i do  believe it's urnelated.
@hashhar @ebyhr @grantatspothero i think Kudu tests reach flakiness level warranting disabling them, or reverting recent test changes.  Can you please take care of this ASAP?
cc @wendigo

File: core/trino-main/src/main/java/io/trino/SystemSessionProperties.java (hunk 1 above)
Comment (raunaqmorarka) [reply core/trino-main/src/main/java/io/trino/SystemSessionProperties.java]: done

**Comment by sopel39** [comment]
The nest step would be extracting system session properties

**Comment by grantatspothero** [comment]
@findepi understand, I'm currently working on improving the flakes

PR here: https://github.com/trinodb/trino/issues/11203
issue here: https://github.com/trinodb/trino/issues/11203

**Comment by mosabua** [comment]
Is this page correct still https://trino.io/docs/current/admin/properties-optimizer.html .. should we do some clean up / move of other properties to that page? wdyt @raunaqmorarka @findepi @hashhar ?

**Comment by raunaqmorarka** [comment]
> Is this page correct still https://trino.io/docs/current/admin/properties-optimizer.html .. should we do some clean up / move of other properties to that page? wdyt @raunaqmorarka @findepi @hashhar ?

This change didn't invalidate any of the existing docs, it just moved configs which were already related to optimizer to a different class. We can potentially document all the configs from the new class in https://trino.io/docs/current/admin/properties-optimizer.html

**Pull request merged**